lib/pc_matrix.c

```diff
@@ -0,0 +1,80 @@
+/***********************************************************************
+* pc_matrix.c
+*
+*  Simple matrix library.
+*  Allow this library to be used both inside and outside a
+*  PgSQL backend.
+*
+*  PgSQL Pointcloud is free and open source software provided
+*  by the Government of Canada
+*  Copyright (c) 2013 Natural Resources Canada
+*  Copyright (c) 2017 Oslandia
+*
+***********************************************************************/
+
+#include "pc_api_internal.h"
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+/**
+* Create the 4x4 matrix associated to the (qw, qx, qy, qz) quaternion. Assume
+* that the quaternion is a unit quaternion.
+*/
+PCMAT44 *
```

Comment:
why do you use a PCMAT44 for a normalized quaternion ? I suggest to use a linear PCMAT33 for a normalized quaternion and a projective PCMAT44 for an unnormalized quaternion, replacing 1's by |q|^2 on the diagonal.